styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorageTransaction.java

```diff
@@ -131,4 +143,34 @@ public WorkflowId patchState(WorkflowId workflowId, WorkflowState state) throws
 
     return workflowId;
   }
+
+  @Override
+  public Optional<PersistentWorkflowInstanceState> activeState(WorkflowInstance instance) throws IOException {
```

Comment:
We could also refactor these methods into e.g. static methods acting on the `DatastoreReaderWriter` interface, which is implemented by both `Transaction` and `Datastore`.

As for always using transactions, it does involve  `beginTransaction` and `commit` RPCs, with associated overhead. Not sure what the performance implications will be beyond that.

The only real use of this method that I could find is by `KubernetesDockerRunner.lookupPodRunState()`, and I think it might be good to move that into a transaction. The current behavior of (outside a transaction) reading the current state from datastore and then emitting events based on that seems inherently racy.